File: java/hcx-integrator-sdk/src/main/java/io/hcxprotocol/utils/UUIDUtils.java
package io.hcxprotocol.utils;

import lombok.experimental.UtilityClass;

/**
 * The UUID Util to validate its format.
 */
@UtilityClass
public class UUIDUtils {
    public static boolean isUUID(String s) {
        return s.matches("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
    }
}

File: dot-net/hcx-integrator-sdk/Io.HcxProtocol/Utils/UUIDUtils.cs
using System.Text.RegularExpressions;

namespace Io.HcxProtocol.Utils
{
    /**
     * Library  : Io.Hcx.Protocol
     * Author   : WalkingTree Technologies
     * Date     : 15-Mar-2023
     * All Rights Reserved. WalkingTree Technologies.
     **/

    /// <summary>
    /// The UUID Util to validate its format.
    /// </summary>
    public static class UUIDUtils
    {
        public static bool IsUUID(this string s)
        {
            var pattern = "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
            return Regex.IsMatch(s, pattern);
        }

    }
}
